app/controllers/templates_controller.rb

```diff
@@ -0,0 +1,42 @@
+class TemplatesController < ApplicationController
+  def index
+    @templates = Template.where active: true
+  end
+
+  def show
+    @template = Template.find params[:id]
+  end
+
+  def new
+    @template = Template.new
+    @widget_types = WidgetType.where(active: true).all
+  end
+
+  def create
+    permitted_params = TemplateParameters.new(params).permit
+    template = Template.create permitted_params
```

Comment:
The issue I had trying to do it that way was that I couldn't add the WidgetTypes in question to the template without saving the template first.